src/__tests__/which_os_test.js

```diff
@@ -0,0 +1,69 @@
+var whichOs = require('../which_os');
+
+describe('whichOs', function () {
+  it('should returns Android as OS', function () {
+    var androidUA = [
+      'Mozilla/5.0 (Linux; U; Android 4.0.3; ko-kr; LG-L160L Build/IML74K) AppleWebkit/534.30 (KHTML, like Gecko) Version/4.0 Mobile Safari/534.30',
+      'Mozilla/5.0 (Linux; U; Android 4.0.3; de-ch; HTC Sensation Build/IML74K) AppleWebKit/534.30 (KHTML, like Gecko) Version/4.0 Mobile Safari/534.30',
+      'Mozilla/5.0 (Linux; U; Android 2.3; en-us) AppleWebKit/999+ (KHTML, like Gecko) Safari/999.9'
+    ].forEach(function (userAgent) {
+      expect(whichOs(userAgent)).to.equal('android');
+    })
+  });
+
+  it('should returns Blackberry as OS', function () {
+    var bbUA = [
+      'Mozilla/5.0 (BB10; <Device Model>) AppleWebKit/<WebKit Version> (KHTML, like Gecko) Version/<BB Version #> Mobile Safari/<WebKit Version>'
+    ].forEach(function (userAgent) {
+      expect(whichOs(userAgent)).to.equal('blackberry');
+    })
+  });
+
+  it('should returns iPhone as OS', function () {
+    var iphoneUA = [
+      'Mozilla/5.0 (iPhone; U; CPU like Mac OS X; en) AppleWebKit/420+ (KHTML, like Gecko) Version/3.0 Mobile/1C25 Safari/419.3',
+      'Mozilla/5.0 (iPhone; U; CPU iPhone OS 4_0 like Mac OS X; en-us) AppleWebKit/532.9 (KHTML, like Gecko) Version/4.0.5 Mobile/8A293 Safari/6531.22.7',
+      'Mozilla/5.0 (iPod touch; CPU iPhone OS 7_0_3 like Mac OS X) AppleWebKit/537.51.1 (KHTML, like Gecko) Version/7.0 Mobile/11B511 Safari/9537.53'
+    ].forEach(function (userAgent) {
+      expect(whichOs(userAgent)).to.equal('iphone');
+    })
+  });
+
+  it('should returns iMac as OS', function () {
```

Comment:
"returns 'mac' as OS"

In this case it would be correct for the library to return 'OS X' as the name of the OS, but let's keep it as 'mac' for now to be consistent with `isMac` function.